extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -0,0 +1,267 @@
+                const connector = new Connector({
+                    name: connectorName,
+                    config,
+                    buckets,
+                    logger: this._logger,
+                    kafkaConnectHost: this._kafkaConnectHost,
+                    kafkaConnectPort: this._kafkaConnectPort,
+                });
+                return connector;
+            }));
+            return connectors;
+        } catch (err) {
+            this._logger.error('An error occured while getting old connectors', {
+                method: 'ConnectorsManager._getOldConnectors',
+                error: err.description || err.message,
+            });
+            throw errors.InternalError.customizeDescription(err.description);
+        }
+    }
+
+    /**
+     * Initialize previously created connector instances and
+     * creates new connectors based on configuration
+     * @returns {Promise|Connector[]} list connectors
+     * @throws {InternalError}
+     */
+    async initializeConnectors() {
+        try {
+            // get and initialize old connectors
+            const oldConnectorNames = await this._kafkaConnect.getConnectors();
+            if (oldConnectorNames) {
+                const oldConnectors = await this._getOldConnectors(oldConnectorNames);
+                this._connectors.push(...oldConnectors);
+                this._oldConnectors.push(...oldConnectors);
+            }
+            // Add connectors if required number of connectors not reached
+            const nbConnectorsToAdd = this._nbConnectors - this._connectors.length;
+            if (nbConnectorsToAdd > 0) {
+                for (let i = 0; i < nbConnectorsToAdd; i++) {
+                    // eslint-disable-next-line no-await-in-loop
+                    const newConnector = await this.addConnector();
+                    this._connectors.push(newConnector);
+                }
+            }
+            return this._connectors;
+        } catch (err) {
+            this._logger.error('An error occured while initializing connectors', {
+                method: 'ConnectorsManager.initializeConnectors',
+                error: err.description || err.message,
+            });
+            throw errors.InternalError.customizeDescription(err.description);
+        }
+    }
+
+    /**
+     * Removes invalid buckets from connector config
+     * @param {Connector} connector connector
+     * @param {string[]} buckets valid bucket names
+     * @returns {Promise|undefined} undefined
+     * @throws {InternalError}
+     */
+    async removeConnectorInvalidBuckets(connector, buckets) {
+        try {
+            // getting connector's invalid buckets
+            const invalidBuckets = connector.buckets.filter(bucket =>
+                !buckets.includes(bucket));
+            if (invalidBuckets.length > 0) {
+                // removing invalid buckets
+                await Promise.all(invalidBuckets.map(bucket =>
```

Comment:
How many invalid buckets could there be at worst? Asking to check if there could be so many (say more than 1000) that it might be a bad idea to do them all in parallel, for all connectors, and instead using `async.eachSeries`, `async.eachLimit` or equivalent could be better.